src/Simulation/CsharpGeneration/SimulationCode.fs

```diff
@@ -1619,13 +1619,15 @@ module SimulationCode =
     /// Builds the SyntaxTree for callables and types loaded via test names,
     /// formats it and returns it as a string.
     /// Returns null if no elements have been loaded via test name.
-    let loadedViaTestNames (dllName : NonNullable<string>) globalContext =
+    let loadedViaTestNames (dllName : NonNullable<string>) (globalContext : CodegenContext) =
         let isLoadedViaTestName nsElement =
-            let asOption = function | Value _ -> Some nsElement | _ -> None
-            match nsElement with
-            | QsCallable c as e -> SymbolResolution.TryGetTestName c.Attributes
-            | QsCustomType t as e -> SymbolResolution.TryGetTestName t.Attributes
-            |> asOption
+            if globalContext.ExposeReferencesViaTestNames then
+                let asOption = function | Value _ -> Some nsElement | _ -> None
```

Comment:
What value does this function operate on? `function` should be short-hand for `match x with`, but I don't know what the "x" is in this case.